utils/ansible.py

```diff
@@ -0,0 +1,281 @@
+from distutils.sysconfig import get_python_lib
+from os import listdir, mkdir, path, remove
+from shutil import copy, copyfile, rmtree
+from subprocess import check_output, CalledProcessError, STDOUT
+
+from fauxfactory import gen_alphanumeric
+from utils import conf
+from utils.providers import get_crud
+
+from git import Repo
+from yaml import load, dump
+
+
+yml_path = path.join(path.dirname(__file__) + "/manageiq_ansible_module/")
+library_path = get_python_lib()
+basic_yml_path = path.dirname(__file__) + "/ansible_conf/"
+library_path_to_copy_to = basic_yml_path + "library"
+providers_basic_script = "providers_basic_script.yml"
+users_basic_script = "users_basic_script.yml"
+basic_script = "_basic_script.yml"
+basic_script_bk = "_basic_script_bk.yml"
+yml = ".yml"
+random_token = str(gen_alphanumeric(906))
+random_miq_user = str(gen_alphanumeric(8))
+local_git_repo = "manageiq_ansible_module"
+pulled_repo_library_path = local_git_repo + "/library"
+remote_git_repo_url = "git://github.com/dkorn/manageiq-ansible-module.git"
+
+
+def fetch_miq_ansible_module():
+    if path.isdir(local_git_repo):
+        rmtree(local_git_repo)
+    mkdir(local_git_repo)
+    if path.isdir(library_path_to_copy_to):
+        rmtree(library_path_to_copy_to)
+    mkdir(library_path_to_copy_to)
+    Repo.clone_from(remote_git_repo_url, local_git_repo)
+    src_files = listdir(pulled_repo_library_path)
+    for file_name in src_files:
+        full_file_name = path.join(pulled_repo_library_path, file_name)
+        if path.isfile(full_file_name):
+            copy(full_file_name, library_path_to_copy_to)
+    rmtree(local_git_repo)
+
+
+def get_values_from_conf(provider, script_type):
```

Comment:
**optional:** i see potential for enhancing those if chains to something supporting long term maintenance, i'm happy to assist now or in future